homeassistant/helpers/config_validation.py

```diff
@@ -687,38 +680,51 @@ def __call__(self, data):
             # This is a legacy config, print warning
             extra_key_errs = [err.path[-1] for err in orig_err.errors
                               if err.error_message == 'extra keys not allowed']
-            if extra_key_errs:
-                msg = "Your configuration contains extra keys " \
-                      "that the platform does not support.\n" \
-                      "Please remove "
-                submsg = ', '.join('[{}]'.format(err) for err in
-                                   extra_key_errs)
-                submsg += '. '
-                if hasattr(data, '__config_file__'):
-                    submsg += " (See {}, line {}). ".format(
-                        data.__config_file__, data.__line__)
-                if hasattr(data, '__configuration_topic__'):
-                    submsg += " (Received on MQTT topic {}). ".format(
-                        data.__configuration_topic__)
-                redacted_data = {}
-
-                for k, v in data.items():
-                    if (any(regex.match(k) for regex in WHITELIST) or
-                            k in extra_key_errs):
-                        redacted_data[k] = v
-                    else:
-                        redacted_data[k] = '<redacted>'
-                submsg += "\n(Offending data: {}".format(
-                    json.dumps(redacted_data))
-
-                msg += submsg
-                logging.getLogger(__name__).warning(msg)
-                INVALID_EXTRA_KEYS_FOUND.append(submsg)
-            else:
+
+            if not extra_key_errs:
                 # This should not happen (all errors should be extra key
                 # errors). Let's raise the original error anyway.
                 raise orig_err
 
+            WHITELIST = [
+                re.compile(CONF_NAME),
+                re.compile(CONF_PLATFORM),
+                re.compile('.*_topic'),
+                ]
+
+            msg = "Your configuration contains extra keys " \
+                  "that the platform does not support.\n" \
+                  "Please remove "
+            submsg = ', '.join('[{}]'.format(err) for err in
+                               extra_key_errs)
+            submsg += '. '
+
+            # Add file+line information, if available
+            if hasattr(data, '__config_file__'):
+                submsg += " (See {}, line {}). ".format(
+                    data.__config_file__, data.__line__)
+
+            # Add MQTT topic information, if available
+            if hasattr(data, '__configuration_topic__'):
```

Comment:
Can we generalize this, maybe name it `__configuration_source__` ?

Reply:
OK, and then it should be a helpful already formatted string instead, for example:
`Received on MQTT topic homeassistant/binary_sensor/0x00158d000244e5ff/occupancy/config`
Instead of:
`homeassistant/binary_sensor/0x00158d000244e5ff/occupancy/config`